internal/images/custom-scorecard-tests/main.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The Cryostat Authors
+//
+// The Universal Permissive License (UPL), Version 1.0
+//
+// Subject to the condition set forth below, permission is hereby granted to any
+// person obtaining a copy of this software, associated documentation and/or data
+// (collectively the "Software"), free of charge and under any and all copyright
+// rights in the Software, and any and all patent rights owned or freely
+// licensable by each licensor hereunder covering either (i) the unmodified
+// Software as contributed to or provided by such licensor, or (ii) the Larger
+// Works (as defined below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+// one is included with the Software (each a "Larger Work" to which the Software
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition:
+// The above copyright notice and either this complete permission notice or at
+// a minimum a reference to the UPL must be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	scapiv1alpha3 "github.com/operator-framework/api/pkg/apis/scorecard/v1alpha3"
+	apimanifests "github.com/operator-framework/api/pkg/manifests"
+
+	tests "github.com/cryostatio/cryostat-operator/internal/test/scorecard"
+)
+
+const podBundleRoot = "/bundle"
+
+func main() {
+	entrypoint := os.Args[1:]
+	if len(entrypoint) == 0 {
+		log.Fatal("specify one or more test name arguments")
+	}
+
+	// Get namespace from SCORECARD_NAMESPACE environment variable
+	namespace := os.Getenv("SCORECARD_NAMESPACE")
+	if len(namespace) == 0 {
+		log.Fatal("SCORECARD_NAMESPACE environment variable not set")
+	}
+
+	// Read the pod's untar'd bundle from a well-known path.
+	bundle, err := apimanifests.GetBundleFromDir(podBundleRoot)
+	if err != nil {
+		log.Fatalf("failed to read bundle manifest: %s", err.Error())
+	}
+
+	var result scapiv1alpha3.TestStatus
+
+	switch entrypoint[0] {
```

Comment:
It's fine for now since there is only one test that can be invoked, but I think we may as well prepare to handle the case where multiple tests exist and multiple can be specified. The `entrypoint := os,Args[1:]` already gets us halfway there - I think basically this `switch` just needs to be iterated over and the index used instead of `[0]`.

Maybe it's also worth doing two passes - the first to make sure all the named tests are known/valid, and the second to actually run them.

Reply:
Great idea!! I just noticed the operator-sdk set up the scorecard tests like:

https://github.com/ebaron/cryostat-operator/blob/5b1501ac567c51b6a176f680985d0104026ddc21/config/scorecard/patches/olm.config.yaml#L1

It seems they launch a separate container (same scorecard image) for each test. Should we follow them too?

Reply:
This is actually what's being done already, I just followed the Operator SDK example. One test per container, which is why the boilerplate in main.go considers only `entrypoint[0]`. That said, there doesn't appear to be any reason in the API why the container can't run more than one test. I'll modify main.go to support this.

Reply:
I reworked the `main.go` to support multiple test arguments now.